File: app/src/main/java/com/example/devicemanager/adapter/SummaryAdapter.java
package com.example.devicemanager.adapter;

import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.devicemanager.R;
import com.example.devicemanager.activity.SummaryListDetailActivity;
import com.example.devicemanager.manager.Contextor;

public class SummaryAdapter extends RecyclerView.Adapter<SummaryAdapter.Holder> {
    private Context context;
    private int[] available,count,total;
    private Context contector = Contextor.getInstance().getContext();
    private String[] brand,type,
            funiture = contector.getResources().getStringArray(R.array.furniture),
            other = contector.getResources().getStringArray(R.array.other_summary);

    public SummaryAdapter(Context context) {
        this.context = context;

    }

    public void setBrand(String[] brand) {
        this.brand = brand;
    }

    public void setTotal(int[] total) {
        this.total = total;
    }

    public void setCount(int[] count) {
        this.count = count;
    }

    public void setAvailable(int[] available) {
        this.available = available;
    }

    public void setType(String[] type) {
        this.type = type;
    }


    @NonNull
    @Override
    public SummaryAdapter.Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_other_item, parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, final int position) {
        holder.setItem(position);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (type.length > 1) {
                    Intent intent = new Intent(context, SummaryListDetailActivity.class);
                    intent.putExtra("Type", type[position]);
                    intent.putExtra("Brand", "-");
                    context.startActivity(intent);
                } else {
                    Intent intent = new Intent(context, SummaryListDetailActivity.class);
                    intent.putExtra("Type", type[0]);
                    intent.putExtra("Brand", brand[position]);
                    context.startActivity(intent);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        if (type == null)
            return 0;
        if (type.length == 1) {
            return brand.length;
        }
        return type.length;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    class Holder extends RecyclerView.ViewHolder {
        TextView tvBrand, tvTotal, tvAvailable, tvActive, tvType;
        ImageView imgView;

        Holder(View itemView) {
            super(itemView);
            tvBrand = itemView.findViewById(R.id.tvBrand);
            tvTotal = itemView.findViewById(R.id.tvTotal);
            tvAvailable = itemView.findViewById(R.id.tvAvailable);
            tvActive = itemView.findViewById(R.id.tvActive);
            tvType = itemView.findViewById(R.id.tvType);
            imgView = itemView.findViewById(R.id.imgView);
        }

        public void setItem(int position) {
            if (brand != null) {
                tvBrand.setText("" + brand[position]);
                switch (brand[position].toLowerCase()) {
                    case "apple":
                        imgView.setImageResource(R.drawable.ic_brand_apple);
                        break;
                    case "hp":
                        imgView.setImageResource(R.drawable.ic_brand_hp);
                        break;
                    case "lenovo":
                        imgView.setImageResource(R.drawable.ic_brand_lenovo);
                        break;
                    case "dell":
                        imgView.setImageResource(R.drawable.ic_brand_dell);
                        break;
                    case "true idc chromebook 11":
                        imgView.setImageResource(R.drawable.ic_brand_chromebook);
                        break;
                    case "-":
                        imgView.setImageResource(R.drawable.ic_barcode);
                        break;
                }
            }
            tvTotal.setText("" + total[position]);
            tvAvailable.setText("" + available[position]);
            tvActive.setText("" + count[position]);
            if (type.length <= 1) {
                tvType.setText(type[0]);
                setImage(0);
            } else {
                tvType.setText("" + type[position]);
                setImage(position);
            }
        }

        private void setImage(int position) {
            if (brand!=null){
                return;
            }
            if(type[position].matches("LAPTOP")){
                imgView.setImageResource(R.drawable.ic_summary_laptop);
                return;
            }
            for (int i = 0; i < other.length; i++) {
                if (other[i].toUpperCase().trim().matches(type[position].toUpperCase().trim())){
                    imgView.setImageResource(R.drawable.ic_summary_other);
                    return;
                }
            }
            for (int i = 0; i < funiture.length; i++) {
                if (funiture[i].toUpperCase().trim().matches(type[position].toUpperCase().trim())){
                    imgView.setImageResource(R.drawable.ic_summary_furniture_s);
                    return;
                }
            }
            imgView.setImageResource(R.drawable.ic_summary_device_s);

        }
    }
}

File: app/src/main/java/com/example/devicemanager/adapter/SummaryAdapter.kt
package com.example.devicemanager.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.example.devicemanager.R
import com.example.devicemanager.activity.SummaryListDetailActivity
import com.example.devicemanager.manager.Contextor

class SummaryAdapter(private val context: Context) : RecyclerView.Adapter<SummaryAdapter.Holder>() {
    private var available: IntArray? = null
    private var count: IntArray? = null
    private var total: IntArray? = null
    private val contector = Contextor.getInstance().context
    private var brand: Array<String>? = null
    private var type: Array<String>? = null
    private val funiture = contector!!.resources.getStringArray(R.array.furniture)
    private val other = contector!!.resources.getStringArray(R.array.other_summary)

    fun setBrand(brand: Array<String>) {
        this.brand = brand
    }

    fun setTotal(total: IntArray) {
        this.total = total
    }

    fun setCount(count: IntArray) {
        this.count = count
    }

    fun setAvailable(available: IntArray) {
        this.available = available
    }

    fun setType(type: Array<String>) {
        this.type = type
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SummaryAdapter.Holder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_other_item, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        holder.setItem(position)
        holder.itemView.setOnClickListener {
            if (type!!.size > 1) {
                val intent = Intent(context, SummaryListDetailActivity::class.java)
                intent.putExtra("Type", type!![position])
                intent.putExtra("Brand", "-")
                context.startActivity(intent)
            } else {
                val intent = Intent(context, SummaryListDetailActivity::class.java)
                intent.putExtra("Type", type!![0])
                intent.putExtra("Brand", brand!![position])
                context.startActivity(intent)
            }
        }
    }

    override fun getItemCount(): Int {
        if (type == null)
            return 0
        return if (type!!.size == 1) {
            brand!!.size
        } else type!!.size
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvBrand: TextView
        var tvTotal: TextView
        var tvAvailable: TextView
        var tvActive: TextView
        var tvType: TextView
        var imgView: ImageView

        init {
            tvBrand = itemView.findViewById(R.id.tvBrand)
            tvTotal = itemView.findViewById(R.id.tvTotal)
            tvAvailable = itemView.findViewById(R.id.tvAvailable)
            tvActive = itemView.findViewById(R.id.tvActive)
            tvType = itemView.findViewById(R.id.tvType)
            imgView = itemView.findViewById(R.id.imgView)
        }

        fun setItem(position: Int) {
            if (brand != null) {
                tvBrand.text = "" + brand!![position]
                when (brand!![position].toLowerCase()) {
                    "apple" -> imgView.setImageResource(R.drawable.ic_brand_apple)
                    "hp" -> imgView.setImageResource(R.drawable.ic_brand_hp)
                    "lenovo" -> imgView.setImageResource(R.drawable.ic_brand_lenovo)
                    "dell" -> imgView.setImageResource(R.drawable.ic_brand_dell)
                    "true idc chromebook 11" -> imgView.setImageResource(R.drawable.ic_brand_chromebook)
                    "-" -> imgView.setImageResource(R.drawable.ic_barcode)
                }
            }
            tvTotal.text = "" + total!![position]
            tvAvailable.text = "" + available!![position]
            tvActive.text = "" + count!![position]
            if (type!!.size <= 1) {
                tvType.text = type!![0]
                setImage(0)
            } else {
                tvType.text = "" + type!![position]
                setImage(position)
            }
        }

        private fun setImage(position: Int) {
            if (brand != null) {
                return
            }
            if (type!![position].matches("LAPTOP".toRegex())) {
                imgView.setImageResource(R.drawable.ic_summary_laptop)
                return
            }
            for (i in other.indices) {
                if (other[i].toUpperCase().trim { it <= ' ' }.matches(type!![position].toUpperCase().trim { it <= ' ' }.toRegex())) {
                    imgView.setImageResource(R.drawable.ic_summary_other)
                    return
                }
            }
            for (i in funiture.indices) {
                if (funiture[i].toUpperCase().trim { it <= ' ' }.matches(type!![position].toUpperCase().trim { it <= ' ' }.toRegex())) {
                    imgView.setImageResource(R.drawable.ic_summary_furniture_s)
                    return
                }
            }
            imgView.setImageResource(R.drawable.ic_summary_device_s)

        }
    }
}
